### PR TITLE
Created a commissaire.util.ssh package.

### PR DESCRIPTION
**Pull request by ashcrow**
This is the first module in the new util package. It adds a common class
for handling temporary ssh keys. It also updates code which
previously manually handled keys.

### CODE REVIEW

**Comment by ashcrow** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;8zmujVM6&#x2F;371-2-multi-host-mgr-merge-temporary-ssh-key-code"> (2) [multi-host-mgr] Merge temporary ssh key code</a>

**Comment by ashcrow** [comment]
I think the CI failure is leftover from the Travis issues.

File: src/commissaire/util/ssh.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+class TemporarySSHKey:
+    """
+    An abstraction for temporary ssh keys.
+    """
+
+    def __init__(self, host, logger):
+        """
+        Initializes an instance of the TemporarySSHKey class.
+
+        :param host: Host to grab ket data from.
```
Comment (mbarnes) [inline src/commissaire/util/ssh.py]: Typo: `ket data` :point_right: `key data`

**Comment by mbarnes** [comment]
Nice looking cleanup.

It occurred to me `TemporarySSHKey` could follow a context manager pattern by adding an `__enter__` method and converting `remove` to `__exit__`.  Mainly just syntactic sugar, but it would be neat to do `with TemporarySSHKey(host, logger) as key` and know it'll get removed.  Just a thought, certainly not a blocker.

**Comment by ashcrow** [comment]
I'll add context manager support just for the heck of it :-)

**Comment by ashcrow** [comment]
:arrow_up:

**Comment by mbarnes** [comment]
Ah, I was expecting to see `with` blocks in the "jobs" code but this works too.  :+1:

**Comment by ashcrow** [comment]
@mbarnes admittedly I am being lazy and not redoing the areas to use the context manager. But later we can utilize it.

**Comment by mbarnes** [comment]
I hear ya.  Auto-squashed and human-merged.

**Pull request closed without merging**